### PR TITLE
Extension Point JSDoc and Live Example Code Embeds in docs (3.1)

### PR DESCRIPTION
**Pull request by jharwig**
- [x] @joeferner
- [x] @diegogrz
- [x] @mwizeman @sfeng88
- [x] @joeybrk372 @rygim @jharwig @EvanOxfeld

Added JSDoc section for extension points, and documented all existing supported points. A new repository [doc-examples](https://github.com/visallo/doc-examples) are live examples of (almost) all extension points. These are then extracted into the docs gitbook build process.

_I haven't completed the tutorials for the last 5 extension points (search toolbar -> web socket), they will go in another PR._

The Visallo JSDoc theme project includes plugins for special handing of extension points. It will search the code for `documentExtensionPoint` calls automatically and import into the output.

**Testing Instructions**: Now testers can use the doc-examples repository to test extension points. View that repos README for instructions.

CHANGELOG
Added: `org.visallo.search.advanced` extension point with tutorial and documentation
Documentation: New JSDoc documentation of front-end API, services, extension points
Documentation: New tutorials in documentation with references to actual plugin projects
Removed: `org.visallo.detail.extensions` extension point, instead use the `org.visallo.layout.component` extension to adjust element inspector display

---

## Rendered PDF of New Docs
[extension-point-book.pdf](https://dl.dropboxusercontent.com/u/7953598/tempShare/extension-point-updates.pdf)

## JSDoc Screenshot

screen shot 2017-02-27 at 1 18 29 pm

### CODE REVIEW

**Comment by jharwig** [comment]
Reminder to cherry-pick to master

File: docs/extension-points/front-end/graphDecorations/index.md.bak [hunk 1]
```diff
@@ -0,0 +1,154 @@
+## Graph Decorations
```
Comment (mwizeman) [inline docs/extension-points/front-end/graphDecorations/index.md.bak]: Should this .bak file be checked in?

Comment (jharwig) [inline docs/extension-points/front-end/graphDecorations/index.md.bak]: nope, thanks

File: docs/extension-points/front-end/searchAdvanced/index.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Using the service created earlier, we can make a data request to run the search and get the result as a promise. Using the public API we access the `List` component for display.
+
+{% github_embed "https://github.com/visallo/doc-examples/blob/d0c25e6a/extension-search-advanced/src/main/resources/org/visallo/examples/search_advanced/example.js#L75-L81" %}{% endgithub_embed %}
+
+The results should be rendered in an element outside of the extension component. The selector is provided as an attribute.
+
+{% github_embed "https://github.com/visallo/doc-examples/blob/d0c25e6a/extension-search-advanced/src/main/resources/org/visallo/examples/search_advanced/example.js#L29" %}{% endgithub_embed %}
+
+The display the results, render the List component into the results element, and enable infiniteScrolling.
```
Comment (jharwig) [inline docs/extension-points/front-end/searchAdvanced/index.md]: Not sure it should be here, but also not sure where.

**Comment by jharwig** [comment]
Wait on merge... making one final edit pass

**Pull request merged**